### DISCUSSION TITLE
Query regarding google::cloud::ExperimentalTag{}

### DISCUSSION THREAD

**Discussion by Rathi-G** [googleapis/google-cloud-cpp#10305] [General]
Hi,
What is the use of AsyncStreamingRecognize(google::cloud::ExperimentalTag{})?
Does is mean the code/functionality is not reliable yet?

Is it a stable code, if not when can we expect the stable version for this?

Is it safe to use it in the product development in commercial purpose?

Is there any other workaround ?

**Comment by dbolduc** [comment]
Hey, thanks for the question.

> What is the use of AsyncStreamingRecognize(google:cloud:ExperimentalTag{})?

As a preface, this API exists in two places:
* `speech::SpeechClient::AsyncStreamingRecognize(...)`
* `speech_v2::SpeechClient::AsyncStreamingRecognize(...)`

The `speech_v2::SpeechClient` corresponds to the [v2 version of the Cloud Speech-to-Text service](https://cloud.google.com/speech-to-text/v2/docs), which is **not** GA. I am going to assume we are talking about the `speech::SpeechClient`, which is GA. (For more details on the different versions of the cloud service, see [here](https://github.com/googleapis/google-cloud-cpp/issues/10285#issuecomment-1321507557)).

> Does is mean the code/functionality is not reliable yet?

The code is reliable, but it is marked as experimental because it is tricky to use. I opened #10307 to remove the experimental tag, because I think it is confusing.

Aside: The API is tricky to use, but we do have some samples to get you started: https://github.com/GoogleCloudPlatform/cpp-samples/tree/main/speech/api

> Is it safe to use it in the product development in commercial purpose?

Yes, but with a caveat. At some point we will change the API signature (e.g. to remove `ExperimentalTag` as a parameter). You would have to make the same change in your code when you update your version of `google-cloud-cpp`.

There will always be some way to invoke the [`StreamingRecognize`](https://github.com/googleapis/googleapis/blob/5686f7b6d29817c590ecac4961183331d0475923/google/cloud/speech/v1/cloud_speech.proto#L69-L71) RPC from the C++ client library.

> Is there any other workaround ?

No.

There are other RPCs that do not involve streaming. (e.g. you could use `SpeechClient::Recognize` to [read from a file](https://github.com/GoogleCloudPlatform/cpp-samples/blob/main/speech/api/transcribe.cc), instead). But there is only one way to use the streaming API.

**Comment by dbolduc** [comment]
Update: We dropped the `ExperimentalTag` from the streaming API (#10340). Hopefully this clears up the confusion. The change is included in our latest release, [v2.5.0](https://github.com/googleapis/google-cloud-cpp/releases/tag/v2.5.0).

https://github.com/googleapis/google-cloud-cpp/blob/45376ec70d1e69730372773fca8c873b904eb0b2/google/cloud/speech/speech_client.h#L198-L201

Thanks for bringing this to our attention.

**Comment by Rathi-G** [comment]
Thank you.
Will update the code and api accordingly.
